src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java

```diff
@@ -36,159 +44,464 @@
+        this.resourceFactory = this.context.getService(jpaRef);
+
+        assertNotNull(this.jdbcFactory);
+        assertNotNull(this.builder);
+        assertNotNull(this.txControl);
+        assertNotNull(this.resourceFactory);
+
+        Properties props = new Properties();
+
+        props.setProperty(JDBC_URL, TEST_DB_URL);
+        props.setProperty(JDBC_USER, "admin");
+        props.setProperty(JDBC_PASSWORD, "admin123");
+
+        DataSource ds = null;
+        try {
+            ds = this.jdbcFactory.createDataSource(props);
+        } catch (SQLException e) {
+            Assert.fail(e.getClass() + " : " + e.getMessage());
+        }
+
+        this.em = this.resourceFactory
+                .getProviderFor(this.builder, singletonMap("javax.persistence.nonJtaDataSource", (Object) ds), null)
+                .getResource(this.txControl);
+
+        assertNotNull(this.em);
+
+        setupDataObjects();
+
+    }
+
+    private void setupDataObjects() {
+        this.inspectionHook = new InspectionHookEntity();
+
+        this.inspectionPort = new InspectionPortEntity();
+
+        this.ingress = new NetworkElementEntity();
+        this.egress = new NetworkElementEntity();
+        this.inspected = new NetworkElementEntity();
+
+        this.ingress.setElementId(IMAC1_STR + IMAC1_STR);
+        this.egress.setElementId(EMAC1_STR + EMAC1_STR);
+        this.inspected.setElementId("iNsPeCtEdPoRt");
+
+        this.ingress.setMacAddresses(asList(IMAC1_STR, IMAC2_STR));
+        this.ingress.setPortIPs(asList(IADDR1_STR, IADDR2_STR));
+
+        this.egress.setMacAddresses(asList(EMAC1_STR, EMAC2_STR));
+        this.egress.setPortIPs(asList(EADDR1_STR, EADDR2_STR));
+
+        this.inspected.setMacAddresses(asList(INSPMAC1_STR));
+
+        this.ingress.setIngressInspectionPort(this.inspectionPort);
+        this.egress.setEgressInspectionPort(this.inspectionPort);
+        this.inspected.setInspectionHook(this.inspectionHook);
+
+        this.inspectionPort.setIngressPort(this.ingress);
+        this.inspectionPort.setEgressPort(this.egress);
+        this.inspectionHook.setInspectedPort(this.inspected);
+
+        this.inspectionHook.setInspectionPort(this.inspectionPort);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+
+        if (this.redirApi != null) {
+            this.redirApi.close();
+        }
+        File dbfile = new File(TEST_DB_FILENAME + ".h2.db");
+
+        if (!dbfile.delete()) {
+            throw new IllegalStateException("Failed to delete database file : " + dbfile.getAbsolutePath());
+        }
+
+        File tracefile = new File(TEST_DB_FILENAME + ".trace.db");
+
+        if (tracefile.exists() &&  !tracefile.delete()) {
+            throw new IllegalStateException("Failed to delete trace file : " + tracefile.getAbsolutePath());
+
+        }
+    }
+
+    @Test
+    public void verifyCorrectNumberOfMacsAdPortIps() throws Exception {
```

Comment:
FYI these unit tests dont follow our typical OSC unit tests guidelines (see  https://github.com/opensecuritycontroller/community/blob/master/development/unit_test_guidelines.md) . Since these are OSGi integration tests rather than regular unit tests we can leave as is though.